src/sentry/tasks/assemble.py

```diff
@@ -284,13 +284,67 @@ def _extract_debug_ids_from_manifest(
     return bundle_id, debug_ids_with_types
 
 
-def _remove_duplicate_artifact_bundles(bundle: ArtifactBundle, bundle_id: str):
+def _bind_or_create_artifact_bundle(
+    bundle_id: uuid,
+    date_added: datetime,
+    org_id: int,
+    archive_file: File,
+    artifact_count: int,
+) -> Tuple[ArtifactBundle, bool]:
+    try:
+        existing_artifact_bundle = ArtifactBundle.objects.get(
+            organization_id=org_id, bundle_id=bundle_id
+        )
+    except ArtifactBundle.DoesNotExist:
+        existing_artifact_bundle = None
+
     with transaction.atomic():
-        # Even though we delete via a QuerySet the associated file is also deleted, because django will still
-        # fire the on_delete signal.
-        ArtifactBundle.objects.filter(
-            ~Q(id=bundle.id), bundle_id=bundle_id, organization_id=bundle.organization_id
-        ).delete()
+        # In case there is not ArtifactBundle with a specific bundle_id, we just create it and return.
+        if existing_artifact_bundle is None:
+            artifact_bundle = ArtifactBundle.objects.create(
```

Comment:
This code path can create duplicates. We still need to have a logic in place to actually get rid of the unintended duplicates.